apps/minifront/src/components/shared/non-native-fee-warning.tsx

```diff
@@ -1,38 +1,105 @@
 import { BalancesResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb.js';
 import { Metadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb.js';
 import { getAssetIdFromValueView } from '@penumbra-zone/getters/value-view';
-import { getAssetId } from '@penumbra-zone/getters/metadata';
 import { useStakingTokenMetadata } from '../../state/shared';
-import { ReactNode } from 'react';
+import { ReactNode, useCallback, useEffect, useState } from 'react';
+import {
+  getAddressIndex,
+  getAmount,
+  getAssetIdFromBalancesResponseOptional,
+} from '@penumbra-zone/getters/balances-response';
+import { viewClient } from '../../clients.ts';
+import { GasPrices } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/fee/v1/fee_pb';
+import { getAssetId } from '@penumbra-zone/getters/metadata';
 
+// Finds the UM token in the user's account balances
 const hasStakingToken = (
+  account: number,
   balancesResponses: BalancesResponse[] = [],
   stakingAssetMetadata?: Metadata,
 ): boolean => {
-  return balancesResponses.some(asset =>
-    getAssetIdFromValueView
-      .optional()(asset.balanceView)
-      ?.equals(getAssetId.optional()(stakingAssetMetadata)),
+  return balancesResponses.some(
+    asset =>
+      getAssetIdFromValueView
+        .optional()(asset.balanceView)
+        ?.equals(getAssetId.optional()(stakingAssetMetadata)) &&
+      getAddressIndex.optional()(asset)?.account === account,
   );
 };
 
-export const useShouldRender = (balancesResponses: BalancesResponse[] = [], amount: number) => {
+// Finds the alt tokens in the user's account balances that can be used for fees
+const hasAltToken = (
```

Comment:
can we combine `hasStakingToken` and `hasAltToken` into [hasTokenBalance](https://github.com/penumbra-zone/web/blob/main/packages/storage/src/indexed-db/index.ts#L872)?

Reply:
refactored the code now + removed `useShouldRender` since there's no need in it after creating `hasTokenBalance`